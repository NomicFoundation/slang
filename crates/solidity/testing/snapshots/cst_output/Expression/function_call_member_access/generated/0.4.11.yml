# This file is generated automatically by infrastructure scripts. Please don't edit by hand.

Source: >
  1  │ x().y                                                                            │ 0..5

Errors: []

Tree:
  - Expression (Rule): # 0..5 "x().y"
      - MemberAccessExpression (Rule): # 0..5 "x().y"
          - FunctionCallExpression (Rule): # 0..3 "x()"
              - PrimaryExpression (Rule): # 0..1 "x"
                  - Identifier (Token): "x" # 0..1
              - FunctionCallOperator (Rule): # 1..3 "()"
                  - ArgumentList (Rule): # 1..3 "()"
                      - OpenParen (Token): "(" # 1..2
                      - CloseParen (Token): ")" # 2..3
          - MemberAccessOperator (Rule): # 3..5 ".y"
              - Period (Token): "." # 3..4
              - Identifier (Token): "y" # 4..5
