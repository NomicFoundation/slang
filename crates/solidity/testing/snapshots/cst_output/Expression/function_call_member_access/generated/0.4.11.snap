# This file is generated automatically by infrastructure scripts. Please don't edit by hand.

Errors: 0

█ Rule(Expression) - (0..5)                       │
└──█ Rule(MemberAccessExpression) - (0..5)        │
   ├──█ Rule(FunctionCallExpression) - (0..3)     │
   │  ├──█ Rule(PrimaryExpression) - (0..1)       │
   │  │  └──► Token(Identifier) - (0..1)          │ x
   │  └──█ Group - (1..3)                         │
   │     └──█ Rule(ArgumentList) - (1..3)         │
   │        ├──► Token(OpenParen) - (1..2)        │ (
   │        └──► Token(CloseParen) - (2..3)       │ )
   └──█ Group - (3..5)                            │
      ├──► Token(Period) - (3..4)                 │ .
      └──► Token(Identifier) - (4..5)             │ y
